moshi/src/main/java/com/squareup/moshi/JsonAdapter.java

```diff
@@ -26,12 +28,25 @@
 public abstract class JsonAdapter<T> {
   public abstract T fromJson(JsonReader reader) throws IOException;
 
+  public final T fromJson(Source source) throws IOException {
+    JsonReader reader = new JsonReader(source);
+    T value = fromJson(reader);
+    reader.close();
+    return value;
+  }
+
   public final T fromJson(String string) throws IOException {
     return fromJson(new JsonReader(string));
   }
 
   public abstract void toJson(JsonWriter writer, T value) throws IOException;
 
+  public final void toJson(Sink sink, T value) throws IOException {
+    JsonWriter writer = new JsonWriter(sink);
+    toJson(writer, value);
+    writer.close();
```

Comment:
Closing is an odd choice. What if you're writing JSON to a file that contains mixed data?